ts/components/TosWebviewComponent.tsx

```diff
@@ -1,94 +1,90 @@
 import * as E from "fp-ts/lib/Either";
 import { pipe } from "fp-ts/lib/function";
-import { Text as NBButtonText } from "native-base";
 import * as React from "react";
 import { useCallback, useState } from "react";
-import { Image, StyleSheet, View, ViewProps } from "react-native";
+import { View, ViewProps } from "react-native";
 import WebView, { WebViewMessageEvent } from "react-native-webview";
 import { WebViewSource } from "react-native-webview/lib/WebViewTypes";
-import { VSpacer } from "@pagopa/io-app-design-system";
-import brokenLinkImage from "../../img/broken-link.png";
+import {
+  FooterWithButtons,
+  H3,
+  IOStyles,
+  Pictogram,
+  VSpacer
+} from "@pagopa/io-app-design-system";
 import I18n from "../i18n";
 import { openWebUrl } from "../utils/url";
 import { AVOID_ZOOM_JS, closeInjectedScript } from "../utils/webview";
-import ButtonDefaultOpacity from "./ButtonDefaultOpacity";
-import FooterWithButtons from "./ui/FooterWithButtons";
+import {
+  trackToSWebViewError,
+  trackToSWebViewErrorRetry
+} from "../screens/authentication/analytics";
+import { FlowType } from "../utils/analytics";
 import { NOTIFY_LINK_CLICK_SCRIPT } from "./ui/Markdown/script";
 import { WebViewMessage } from "./ui/Markdown/types";
-import { H2 } from "./core/typography/H2";
 
 type Props = {
   webViewSource: WebViewSource;
   handleLoadEnd: () => void;
   handleReload: () => void;
   onAcceptTos?: () => void;
-  onExit?: () => void;
   shouldRenderFooter?: boolean;
+  flow: FlowType;
 } & Pick<ViewProps, "testID">;
 
-const styles = StyleSheet.create({
-  errorContainer: {
-    padding: 20,
-    flex: 1,
-    justifyContent: "center",
-    alignItems: "center"
-  },
-  errorButtonsContainer: {
-    position: "absolute",
-    bottom: 30,
-    flex: 1,
-    flexDirection: "row"
-  }
-});
-
 const TosWebviewComponent: React.FunctionComponent<Props> = ({
   handleLoadEnd,
   handleReload,
   webViewSource,
   shouldRenderFooter,
-  onExit,
-  onAcceptTos
+  onAcceptTos,
+  flow
 }: Props) => {
   const [hasError, setHasError] = useState(false);
 
   const handleError = useCallback(() => {
     handleLoadEnd();
     setHasError(true);
-  }, [setHasError, handleLoadEnd]);
+    trackToSWebViewError(flow);
+  }, [handleLoadEnd, flow]);
 
   const handleRetry = useCallback(() => {
     handleReload();
     setHasError(false);
-  }, [setHasError, handleReload]);
+    trackToSWebViewErrorRetry(flow);
+  }, [handleReload, flow]);
 
   const renderError = () => (
```

Comment:
For me, this is not a good idea.  The error handling and reload function are passed from outside, but the logic that is common, i.e. error handling and event tracking, are added inside the component.

Reply:
Then, why not pass them again as props. I think it's always a better approach to decouple components. What do you think?

Reply:
> Then, why not pass them again as props. I think it's always a better approach to decouple components. What do you think?

It is not clear to me why I should do this.. If you can explain to me why I should do as you are suggesting, maybe I can understand the suggestion.
From my point of view, I would agree with you _if_ you were making a _generic component_ to be used in various different cases, but this is a component that redirects the ToS screen, so the tracking function and the logic for handling the error will always be the same. For me, doing it the way you propose would create more complex code because of the handling of the error status outside the component, and I would be passing data as props that are the same in all use cases.

Reply:
I suggest to keep the error component outside the render function of the main component this way, for example

```ts
const ErrorComponent = (props: { handleRetry: () => void }) => {
  const retryButtonLabel = I18n.t("global.buttons.retry");
  const { handleRetry } = props;
  return (
    <>
      <View
        style={[
          IOStyles.flex,
          IOStyles.alignCenter,
          IOStyles.horizontalContentPadding,
          IOStyles.centerJustified
        ]}
        testID="toSErrorContainerView"
      >
        <Pictogram name="stopSecurity" />
        <VSpacer size={8} />
        <H3 testID="toSErrorContainerTitle" style={{ textAlign: "center" }}>
          {I18n.t("onboarding.tos.error")}
        </H3>
      </View>
      <ContentWrapper>
        <ButtonSolid
          fullWidth
          onPress={handleRetry}
          label={retryButtonLabel}
          accessibilityLabel={retryButtonLabel}
          testID="RetryButtonTest"
        />
      </ContentWrapper>
    </>
  );
};
```

Leve it inside the same file, sure, but taking it outside the render function, imho, is a better approach. What do you think?

Reply:
done! 🚀